2022/incoming/time-to-open-presents.pod

```diff
@@ -0,0 +1,170 @@
+Author: oodler@cpan.org
+Title: Wake up! Time to open presents! 
+Topic: $SIG::ALRM, signal handlers, alarm , Try::ALRM
+
+=encoding utf8
+
+=head1 Wake up! Time to open presents!
+
+Clark W. Grisman faced a dilemma, it was the night before
+Christmas; and he found himself up against a programming
+deadline. Why a company at the cutting edge of milk proof
+cereal coatings required him to write a Perl program,
+Clark can't remember. But he does know that he's trying
+to read from a socket that keeps timing out! 
+
+=head1 Traditional C<alarm> Handling
+
+In the past whenever Clark wanted to call a procedure inside
+his Perl code that might take too long to return, he always
+simply copied the example in C<perldoc -f alarm>, which at 
+the time of this Advent season looks like the following:
+
+    # ...
+    
+    eval {
+        local $SIG{ALRM} = sub { die "alarm\n" }; # NB: \n required
+        alarm $timeout;
+        my $nread = sysread $socket, $buffer, $size;
+        alarm 0;
+    };
+    if ($@) {
+        die unless $@ eq "alarm\n";   # propagate unexpected errors
+        # timed out
+    }
+    else {
+        # didn't
+    }
+
+Or if he didn't need it quite so fancy, he'd just nip it down to:
+
+    $SIG{ALRM} = sub { die "alarm\n" }; # NB: \n required
+    alarm $timeout;
+    my $nread = sysread $socket, $buffer, $size;
+    alarm 0;
+
+Clark never liked using C<alarm> in this way because it didn't feel
```

Comment:
```suggestion
Clark never liked using C<alarm> in this latter way because it didn't feel
```

Reply:
I would like you to precise because I suppose only second snippet is not considered "perlish"